### PR TITLE
[rush] Prefer frozen lockfile during `rush update` to minimize lockfile churn

### LINKED ISSUE #2525: [rush] Use `--prefer-frozen-lockfile` in `rush update` (without `--full`)
## Summary

When running `rush update`, I expect it to only change the lockfile if necessary to satisfy new constraints from `rush.json`, `package.json`, `pnpmfile.js`, or `common-versions.json`.

However, the current command runs `pnpm install --no-prefer-frozen-lockfile`, which is a bit more permissive of lockfile updates.

## Repro steps

Pull latest `main` branch of a Rush monorepo, run `rush update`

  **Expected result:** 
No changes detected, lockfile remains the same and does equivalent of `rush install`

  **Actual result:** 
Updates made to pnpm-lock.yaml and repo-state.json

## Details
Change `--no-prefer-frozen-lockfile` to `--prefer-frozen-lockfile`, at least for pnpm >= 5 on workspaces, or just behind an experiment/pnpm config setting.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.40.3 |
| `rushVersion` from rush.json? | 5.40.3 |
| `useWorkspaces` from rush.json? | true |
| Operating system? | Linux |
| Would you consider contributing a PR? | Yes |
| Node.js version (`node -v`)? | 10.24.0 |

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary

Fixes #2525

## Details
Switches pnpm to use `--prefer-frozen-lockfile` for `rush update` when `useWorkspaces` is `true`

## How it was tested
Verified locally on rushstack repo by running `rush update` that no change occurs if the lockfile is up to date, and that modifications occur if it is not.

### CODE REVIEW

**Review by D4N14L — APPROVED** [review]

File: apps/rush-lib/src/logic/base/BaseInstallManager.ts [hunk 1]
```diff
@@ -502,6 +502,9 @@ export abstract class BaseInstallManager {
         } else {
           args.push('--frozen-shrinkwrap');
```
Comment (D4N14L) [inline apps/rush-lib/src/logic/base/BaseInstallManager.ts:503]: Should we update this terminology?

File: apps/rush-lib/src/logic/base/BaseInstallManager.ts [hunk 2]
```diff
@@ -502,6 +502,9 @@ export abstract class BaseInstallManager {
...
         }
+      } else if (this._rushConfiguration.pnpmOptions.preferFrozenLockfileForUpdate) {
+        // In workspaces, we want to avoid unnecessary lockfile churn
```
Comment (D4N14L) [inline apps/rush-lib/src/logic/base/BaseInstallManager.ts:506]: You may also want to condition this option on workspaces support, though opting in does imply that they know what they're doing

**Review by sachinjoseph — APPROVED** [review]

File: apps/rush-lib/src/schemas/rush.schema.json [hunk 3]
```diff
@@ -103,6 +103,10 @@
         "environmentVariables": {
           "$ref": "#/definitions/environmentVariables"
         },
+        "preferFrozenLockfileForUpdate": {
+          "description": "If treu, the for \"rush update\", pnpm will use the --prefer-frozen-lockfile flag to minimize shrinkwrap changes.",
```
Comment (D4N14L) [inline apps/rush-lib/src/schemas/rush.schema.json:107]: treu -> true

File: apps/rush-lib/src/logic/base/BaseInstallManager.ts (hunk 2 above)
Comment (sachinjoseph) [reply apps/rush-lib/src/logic/base/BaseInstallManager.ts:506]: If we condition it on workspaces, then the flag name would also need to imply that - something like preferFrozenLockfileForUpdateInWorkspaces. But like you said, opting in does imply that they know what they're doing, so IMO, this should not be conditioned on workspaces support.

**Review by sachinjoseph — APPROVED** [review]

File: apps/rush-lib/src/logic/base/BaseInstallManager.ts (hunk 1 above)
Comment (dmichon-msft) [reply apps/rush-lib/src/logic/base/BaseInstallManager.ts:503]: Not related.

**Comment by iclanton** [comment]
Would we want this to eventually become the default behavior? If so, should this be an experiment instead of a `rush.json` option?

**Comment by D4N14L** [comment]
> Would we want this to eventually become the default behavior? If so, should this be an experiment instead of a `rush.json` option?

Yes, I think this would be the case (especially since we are planning on workspaces becoming the default as well)

**Comment by dmichon-msft** [comment]
@iclanton, @D4N14L, switch to an experiment from a pnpmOptions value.

**Review by iclanton — APPROVED** [review]

**Pull request merged**